uaclient/cli.py

```diff
@@ -209,41 +209,38 @@ def action_attach(args, cfg):
         print(ua_status.MESSAGE_NONROOT_USER)
         return 1
     contract_client = contract.UAContractClient(cfg)
-    bound_macaroon = None
     if not args.token:
-        bound_macaroon_bytes = sso.discharge_root_macaroon(
-            contract_client)
+        bound_macaroon = None
+        bound_macaroon_bytes = sso.discharge_root_macaroon(contract_client)
         if bound_macaroon_bytes is not None:
```

Comment:
I think this could be simplified now that we don't need `bound_macaroon` outside of this `if` branch:

```py
bound_macaroon_bytes = sso.discharge_root_macaroon(contract_client)
if bound_macaroon_bytes is None:
    print('Could not attach machine. Unable to obtain authenticated user token')
    return 1
bound_macaroon = bound_macaroon_bytes.decode('utf-8')
```

Reply:
Fixed with your suggestion here too. Thanks Dan.